packages/dev/core/src/node.ts

```diff
@@ -974,3 +981,18 @@ export class Node implements IBehaviorAware<Node> {
         };
     }
 }
+
+// TODO: refactor this so the change of this tag can be observed.
+/**
+ * Define an interface for a node to indicate it's info for accessibility.
+ */
+ export interface IAccessibilityTag {

```

Comment:
I believe this interface should look a little different.

As I mentioned in a different comment - the isSalient flag is redundant - if there is no accessibility tag this node should not be considered "accessible".
It should also extend to more than description. For example, it could have an `aria` object that will support the ARIA standard and will allow defining labels, live, ans so on. Something like this:

```javascript
{
  description?: string;
  role?: string;
  aria?: {
    label?: string;
    live?: string;
    relevant?: string;
    // further aria that we want to support
  }
}
```

this object fits 1-to-1 with an HTML object, so you could technically extend an HTML element with these details. You could argue that instead of an `aria` object we should have the actual tag names (like `aria-label` and so on), and that would be fine as well.
This way you provide the developer with everything they need to make the scene fully accessible according to their standards.

This could also be a simple `{[key:string]: string}` map that has its keys automatically added to the corresponding HTML element. So setting:

```javascript
{
  "aria-label": "This is a label",
  "aria-live": "polite"
}
```

will generate the HTML:

```html
<div aria-label="This is a label" aria-live="polite"/> 
```

Reply:
Interesting! That's a good call! I'll look into that more and see how to incorporate it into the code.